### ISSUE TITLE
alignment surprise when displaying tab-completions

### ISSUE THREAD

**Issue by aryairani** [judah/haskeline#114]
![image](https://user-images.githubusercontent.com/538571/53698291-a398a200-3da8-11e9-883f-8b223d910007.png)

I tried using ANSI sequences in the `display` field, but it threw off the alignment of the presented completions:

![image](https://user-images.githubusercontent.com/538571/53698297-cd51c900-3da8-11e9-96f6-0d9a8bdc5530.png)

I naively tried inserting a call to `stringToGraphemes` in `Completion.makeLines` but realized that's a different sort of thing.  What do you think about some way to strip out escape sequences or to provide a length directly?

**Comment by judah** [comment]
Using `stringToGraphemes` inside of `makeLines` seems like a reasonable workaround that would parallel how we handle prefixes.  (Relevant link, though you may have already seen it: https://github.com/judah/haskeline/wiki/ControlSequencesInPrompt)

I think you could convert the `display` completions from strings to lists of graphemes, compute their lengths and pad and concatenate each line, then convert the lines back to strings right before outputting them.  Did you run into problems with that approach?

**Comment by aryairani** [comment]
No; I gave up immediately when `length . stringToGraphemes` didn't give me what I wanted, but with this encouragement I will try again. :)

**Comment by aryairani** [comment]
I understand a little better now, but the ANSI sequences generated by [ansi-terminal](http://hackage.haskell.org/package/ansi-terminal-0.9) aren't bracketed by `'\STX'`, so `stringToGraphemes` doesn't seem to help.

**Comment by aryairani** [comment]
I know the unformatted string (and consequently its length) when I construct the `Completion`, but it's not available at the point that we're ready to `display` it.
